core/src/components/input/input.tsx

```diff
@@ -407,11 +407,10 @@ export class Input implements ComponentInterface {
 
     this.value = '';
 
-    /**
-     * Clearing the text input is considered a user committed change
-     * and should emit a value change event.
-     */
-    this.emitValueChange();
+    this.ionInput.emit({
+      ...ev,
+      data: this.value,
```

Comment:
I'm comparing this with the native behavior in Chrome, and noticed that it does not pass `data` when pressing the "X" button. It only seems to pass `data` when typing. Should we align the behavior here? One thing to note is that the native event is of type `Event`, not `InputEvent`.

CodePen: https://codepen.io/liamdebeasi/pen/oNdXJrp